happi/cli.py

```diff
@@ -191,7 +196,7 @@ def search_parser(
                     range_list = new_range_list
                 else:
                     # subsequent ranges, only take intersection
-                    range_list = set(new_range_list) & set(range_list)
+                    range_list = list(set(new_range_list) & set(range_list))
```

Comment:
super nit: maybe we can hold off on listifying the range_list (and rename it to range_set?) until we return it. It probably doesn't actually matter but it seems wasteful to convert this to a list here and back to a set for the final_results in line 240.

Reply:
I actually think we have to list-ify it here early, since there's a condition where we return `range_list` without modification.  (in the case where there is no `regex_list` to consider)

Reply:
I mean, we could `return list(range_set)` instead

Reply:
Ah yes, fair enough.  I'll be less lazy here